src/main/java/org/takes/facets/auth/codecs/CcAES.java

```diff
@@ -44,77 +43,56 @@
  * AES codec which supports 128 bits key.
  *
  * <p>The class is immutable and thread-safe.
- *
  * @author Jason Wong (super132j@yahoo.com)
  * @version $Id$
  * @since 0.13.8
  */
-@EqualsAndHashCode(of = { "origin", "secret", "enc", "dec" })
+@EqualsAndHashCode(of = {"origin", "key"})
 public final class CcAES implements Codec {
```

Comment:
@sebing please try marking this class as `@Immutable` to prevent non-compliant solutions